File: Back-end/src/main/java/KJX/KJX/entity/Event.java
package KJX.KJX.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "events", indexes = {
    @Index(name = "idx_events_date", columnList = "date"),
    @Index(name = "idx_events_type", columnList = "event_type"),
    @Index(name = "idx_events_team_id", columnList = "team_id")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Event {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 255)
    private String title;
    
    @Column(columnDefinition = "NVARCHAR(MAX)")
    private String description;
    
    @Column(name = "event_type", nullable = false, length = 100)
    private String type; // "match", "scrim", "fanmeet", "content"
    
    @Column(nullable = false)
    private LocalDateTime date;
    
    @Column(length = 10)
    private String time;
    
    @Column(length = 255)
    private String location;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;
    
    @Column(length = 255)
    private String opponent;
    
    @Column(length = 50)
    @Builder.Default
    private String status = "SCHEDULED"; // "SCHEDULED", "COMPLETED", "CANCELLED"
    
    @Column(name = "created_at", updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column(name = "updated_at")
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: nodejs-backend/src/models/Event.js
/**
 * Event Model
 * Converted from Spring Boot JPA Entity
 */

const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Event = sequelize.define('Event', {
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true
    },
    title: {
      type: DataTypes.STRING(255),
      allowNull: false,
      validate: {
        notEmpty: true
      }
    },
    description: {
      type: DataTypes.TEXT
    },
    type: {
      type: DataTypes.STRING(100),
      allowNull: false,
      field: 'event_type'
    },
    date: {
      type: DataTypes.DATE,
      allowNull: false
    },
    time: {
      type: DataTypes.STRING(10)
    },
    location: {
      type: DataTypes.STRING(255)
    },
    team_id: {
      type: DataTypes.BIGINT,
      field: 'team_id',
      references: {
        model: 'teams',
        key: 'id'
      }
    },
    opponent: {
      type: DataTypes.STRING(255)
    },
    status: {
      type: DataTypes.STRING(50),
      defaultValue: 'SCHEDULED',
      validate: {
        isIn: [['SCHEDULED', 'COMPLETED', 'CANCELLED']]
      }
    },
    created_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW,
      field: 'created_at'
    },
    updated_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW,
      field: 'updated_at'
    }
  }, {
    tableName: 'events',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at',
    indexes: [
      {
        fields: ['date'],
        name: 'idx_events_date'
      },
      {
        fields: ['event_type'],
        name: 'idx_events_type'
      },
      {
        fields: ['team_id'],
        name: 'idx_events_team_id'
      }
    ]
  });

  return Event;
};
